akka-stream/src/main/scala/akka/stream/impl/io/FileSubscriber.scala

```diff
@@ -66,13 +70,13 @@ private[akka] class FileSubscriber(f: File, bytesWrittenPromise: Promise[Long],
         chan.force(true)
       } catch {
         case ex: Exception ⇒
-          bytesWrittenPromise.failure(ex)
+          completionPromise.success(IOResult(bytesWritten, Failure(ex)))
       }
       context.stop(self)
   }
 
   override def postStop(): Unit = {
-    bytesWrittenPromise.trySuccess(bytesWritten)
+    completionPromise.trySuccess(IOResult(bytesWritten, Success(Done)))
```

Comment:
we discussed it somewhere in a ticket, that the completing of the promise should be done after the channel as been closed – as closing it may still fail actually (line 81 here)